### ISSUE TITLE
Add mapObject like in OL2

### ISSUE THREAD

**Issue by tds4u** [mapgears/ol3-google-maps#30]
For a working routing application we need access to the former mapObject variable. Can you add such a function so that from the layer object we can receive the Google Maps object?

---

baseMap = new olgm.layer.Google({
    title: "Google", name: "google", visible: true
});
baseMap->getMapObject(); OR baseMap->getGoogleObject();

**Comment by adube** [comment]
Actually, in ol3 there is no way to get the map in which a layer is added from a layer.

You should be able to easily have access to your ol3 map in your app.  For the Google Maps map coming from OLGM, it would need to be exposed with a getter method, as you suggest, but I'd add it to the `olgm.OLGoogleMaps`.  If you're willing to work on a PR, I'd gladly review it.

**Comment by tds4u** [comment]
Trying to compile but:

```
Error: Cannot find module '/scripts/ol3-google-maps/ol3/tasks/generate-externs.js'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3
make: *** [ol3/build/ol-externs.js] Fehler 8
```

Which additional libraries are needed?
//edit: git clone --recursive https://github.com/mapgears/ol3-google-maps.git ;-)

Next error:

```
node build/build.js build/ol3gm.json dist/ol3gm.js
info ol3-google-maps Parsing dependencies
info ol3-google-maps Compiling 39 sources
ERR! compile execvp(): No such file or directory
ERR! compile

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: spawn ENOENT
    at errnoException (child_process.js:988:11)
    at Process.ChildProcess._handle.onexit (child_process.js:779:34)
make: *** [dist/ol3gm.js] Fehler 8
```

//edit: sudo apt-get install default-jre

Hope I can help somebody with this to resolve such problems.

**Comment by adube** [comment]
You need to install the dev environment, which is the same as ol3-cesium: https://github.com/openlayers/ol3-cesium#building-the-library

HTH

**Comment by tds4u** [comment]
So, with debug js all is functional like in OL2. I can access it via olGM.gmap and olGM is

```
olGM = new olgm.OLGoogleMaps({map: olMap}); // map is the ol.Map instance
```

But when using minified js then the .gmap and also .get("gmap" ) ist not available. I'm new to this obfuscation. Please help.

**Comment by adube** [comment]
Methods that contain `@api` in their comment block are exported in the minimized build version.  I.e., those method become available.

`.gmap` is a protected property, so only the class where it was created and its child classes can access it.  Because of that, it gets automatically renamed in the minimized version by the `Closure Compiler`,

The way I suggest to make it available is to create a public method in the `olgm.OLGoogleMaps`, something like `getGoogleMapsMap` and mark it as `@api`.  That way, that method will be available in the minimized version and you'll be able to do `olGM.getGoogleMapsMap()` to get the gmap object.

See https://github.com/mapgears/ol3-google-maps/blob/master/src/ol3googlemaps.js for examples.

**Comment by adube** [comment]
@tds4u Any advancement on this issue ?

**Comment by adube** [comment]
Fixed in #48.
